exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -1023,11 +1883,116 @@ func (e *Exchange) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp curre
 	switch a {
 	case asset.Spot:
 		cp.Delimiter = currency.UnderscoreDelimiter
-		return poloniexAPIURL + tradeSpot + cp.Upper().String(), nil
+		return apiURL + tradeSpotPath + cp.Upper().String(), nil
 	case asset.Futures:
 		cp.Delimiter = ""
-		return poloniexAPIURL + tradeFutures + cp.Upper().String(), nil
+		return apiURL + tradeFuturesPath + cp.Upper().String(), nil
+	default:
+		return "", fmt.Errorf("%w: %q", asset.ErrNotSupported, a)
+	}
+}
+
+// WebsocketSubmitOrder submits an order to the exchange via a websocket connection
+func (e *Exchange) WebsocketSubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	if s.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, s.AssetType)
+	}
+	var stpMode string
+	switch s.TimeInForce {
+	case order.PostOnly:
+		stpMode = "EXPIRE_MAKER"
+	case order.GoodTillCancel:
+		stpMode = "EXPIRE_TAKER"
+	}
+	response, err := e.WsCreateOrder(ctx, &PlaceOrderRequest{
+		Symbol:                  s.Pair,
+		Price:                   s.Price,
+		Amount:                  s.Amount,
+		AllowBorrow:             false,
+		Type:                    orderType(s.Type),
+		Side:                    s.Side.String(),
+		TimeInForce:             timeInForce(s.TimeInForce),
+		ClientOrderID:           s.ClientOrderID,
+		SlippageTolerance:       "0",
+		SelfTradePreventionMode: stpMode,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return s.DeriveSubmitResponse(response.ID)
+}
+
+// WebsocketCancelOrder cancels an order via the websocket connection
+func (e *Exchange) WebsocketCancelOrder(ctx context.Context, req *order.Cancel) error {
+	if req.OrderID == "" && req.ClientOrderID == "" {
+		return order.ErrOrderIDNotSet
+	}
+	if err := req.Validate(req.StandardCancel()); err != nil {
+		return err
+	}
+	resp, err := e.WsCancelMultipleOrdersByIDs(ctx, []string{req.OrderID}, []string{req.ClientOrderID})
+	if err != nil {
+		return err
+	}
+	if len(resp) != 1 {
+		return common.ErrNoResponse
+	} else if resp[0].Code != 0 {
+		return fmt.Errorf("%w: code: %d message: %s", common.ErrInvalidResponse, resp[0].Code, resp[0].Message)
+	}
+	return nil
+}
+
+// orderTypeString return a string representation of order type
+func orderTypeString(oType order.Type) (string, error) {
+	switch oType {
+	case order.Market, order.Limit, order.LimitMaker:
+		return oType.String(), nil
+	case order.StopLimit:
+		return "STOP_LIMIT", nil
+	case order.TrailingStopLimit:
+		return "TRAILING_STOP_LIMIT", nil
+	case order.AnyType, order.UnknownType:
+		return "", nil
+	}
+	return "", fmt.Errorf("%w: %q", order.ErrUnsupportedOrderType, oType)
+}
+
+// StringToOrderType returns an order.Type instance from string
+func StringToOrderType(oTypeString string) order.Type {
```

Comment:
I do not like this with a default of `limit`. Why not use the `order.StringToOrderType`? Your comment doesn't really explain why this is needed. Any strings needed to convert here could be converted by the orders function. If there is an exception, put that in here, else use the order package function